### PR TITLE
[[ Bug 15946 ]] Reinstate array building of CGI control value (POST a…

### PR DESCRIPTION
**Pull request by livecodesebastien**
…nd GET)

The arrays can contain subkeys, and taking this into consideration got removed during the refactoring.

### CODE REVIEW

File: engine/src/srvcgi.cpp [hunk 1]
```diff
@@ -595,11 +595,125 @@ static void cgi_fetch_valueref_for_key(MCVariable *p_variable, MCNameRef p_key,
...
+            return false;
+    }
+
+    // Store the value at the built key path - setvalueref will take care of
+    //  creating subarrays if needed.
+    bool t_success;
+    t_success = p_variable -> setvalueref(t_path . Ptr(), t_path . Size(), false, p_value);
+
+    // Clean keys array
+    t_path . Delete();
```
Comment (peter-b) [inline engine/src/srvcgi.cpp]: I thought an `MCAutoArray()` automatically self-deletes? I don't think this (or the `t_success` variable) is necessary...?

Comment (livecodesebastien) [inline engine/src/srvcgi.cpp]: You're right. Each created `MCNameRef` pushed onto the AutoArray needs to be deleted though (that's what Clean keys array should do).

**Comment by peter-b** [comment]
@livecode-vulcan review ok 33cffae

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 33cffae1a68b260d76cfa1b47d22348ba45f3ce5

**Comment by livecode-vulcan** [comment]
😎 test success 33cffae1a68b260d76cfa1b47d22348ba45f3ce5
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/516)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/520)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/515)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/515)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/524)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/518)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/526)

**Pull request merged**